app/logic/courseManagement.py

```diff
@@ -49,33 +50,36 @@ def updateCourse(courseData):
     """
-    course= Course.get_by_id(courseData['courseID'])
-    for toggler in ["regularOccurenceToggle", "slSectionsToggle", "permanentDesignation"]:
-        courseData.setdefault(toggler, "off")
+    try:
+        course= Course.get_by_id(courseData['courseID'])
+        for toggler in ["regularOccurenceToggle", "slSectionsToggle", "permanentDesignation"]:
+            courseData.setdefault(toggler, "off")
 
-    Course.update(
-        courseName=courseData["courseName"],
-        courseAbbreviation=courseData["courseAbbreviation"],
-        courseCredit=courseData["credit"],
-        isRegularlyOccuring=("on" in courseData["regularOccurenceToggle"]),
-        term=courseData['term'],
-        status=CourseStatus.SUBMITTED,
-        isAllSectionsServiceLearning=("on" in courseData["slSectionsToggle"]),
-        serviceLearningDesignatedSections=courseData["slDesignation"],
-        isPermanentlyDesignated=("on" in courseData["permanentDesignation"]),
-    ).where(Course.id == course.id).execute()
+        Course.update(
+            courseName=courseData["courseName"],
+            courseAbbreviation=courseData["courseAbbreviation"],
+            courseCredit=courseData["credit"],
+            isRegularlyOccuring=("on" in courseData["regularOccurenceToggle"]),
+            term=courseData['term'],
+            status=CourseStatus.SUBMITTED,
+            isAllSectionsServiceLearning=("on" in courseData["slSectionsToggle"]),
+            serviceLearningDesignatedSections=courseData["slDesignation"],
+            isPermanentlyDesignated=("on" in courseData["permanentDesignation"]),
+        ).where(Course.id == course.id).execute()
+        for i in range(1, 7):
+            (CourseQuestion.update(questionContent=courseData[f"{i}"])
+                        .where((CourseQuestion.questionNumber == i) &
+                               (CourseQuestion.course==course)).execute())
 
-    for i in range(1, 7):
-        (CourseQuestion.update(questionContent=courseData[f"{i}"])
-                    .where((CourseQuestion.questionNumber == i) &
-                           (CourseQuestion.course==course)).execute())
-
-    instructorList = []
-    if 'instructor[]' in courseData:
-        instructorList = courseData.getlist('instructor[]')
+        instructorList = []
+        if 'instructor[]' in courseData:
+            instructorList = courseData.getlist('instructor[]')
 
-    CourseInstructor.delete().where(CourseInstructor.course == course).execute()
-    for instructor in instructorList:
-        CourseInstructor.create(course=course, user=instructor)
+        CourseInstructor.delete().where(CourseInstructor.course == course).execute()
+        for instructor in instructorList:
+            CourseInstructor.create(course=course, user=instructor)
+        createLog(f"Edited SLC proposal: {courseData['courseName']}")
+    except:
+        return False;
 
-    return Course.get_by_id(course.id)
+    return course
```

Comment:
You've broken a recent fix in development by changing this line (by taking the old function from the merge conflict, really, instead of the new one). update course needs to return a course object with the updated data, not the original one that was passed in